src/main/java/org/elasticsearch/search/aggregations/reducers/movavg/MovAvgReducer.java

```diff
@@ -118,21 +118,26 @@ public InternalAggregation reduce(InternalAggregation aggregation, ReduceContext
             Double thisBucketValue = resolveBucketValue(histo, bucket, bucketsPaths()[0], gapPolicy);
             currentKey = bucket.getKey();
 
+            // Default is to reuse existing bucket.  Simplifies the rest of the logic,
+            // since we only change newBucket if we can add to it
+            InternalHistogram.Bucket newBucket = bucket;
+
             if (!(thisBucketValue == null || thisBucketValue.equals(Double.NaN))) {
                 values.offer(thisBucketValue);
 
-                double movavg = model.next(values);
-
-                List<InternalAggregation> aggs = new ArrayList<>(Lists.transform(bucket.getAggregations().asList(), FUNCTION));
-                aggs.add(new InternalSimpleValue(name(), movavg, formatter, new ArrayList<Reducer>(), metaData()));
-                InternalHistogram.Bucket newBucket = factory.createBucket(currentKey, bucket.getDocCount(), new InternalAggregations(
-                        aggs), bucket.getKeyed(), bucket.getFormatter());
-                newBuckets.add(newBucket);
+                // Some models (e.g. HoltWinters) have certain preconditions that must be met
+                if (model.hasNext(values.size())) {
+                    double movavg = model.next(values);
 
-            } else {
-                newBuckets.add(bucket);
+                    List<InternalAggregation> aggs = new ArrayList<>(Lists.transform(bucket.getAggregations().asList(), FUNCTION));
+                    aggs.add(new InternalSimpleValue(name(), movavg, formatter, new ArrayList<Reducer>(), metaData()));
+                    newBucket = factory.createBucket(currentKey, bucket.getDocCount(), new InternalAggregations(
+                            aggs), bucket.getKeyed(), bucket.getFormatter());
+                }
             }
 
+            newBuckets.add(newBucket);
+
             if (predict > 0) {
                 if (currentKey instanceof Number) {
                     interval = Math.min(interval, ((Number) bucket.getKey()).longValue() - lastKey);
```

Comment:
Probably for a different PR but the Rounding class is now exposed on the Histogram Agg so we can now use that to work out what the key for the next bucket should be